### PR TITLE
Stop ignoring various logger failures

### PR DESCRIPTION
**Pull request by marc-hb**
2 commits, see commit messages.

Green failures episode "the logger"

Incomplete fix for #297 , more coming.

### CODE REVIEW

File: case-lib/hijack.sh [hunk 1]
```diff
@@ -30,12 +30,23 @@ function func_exit_handler()
...
             sleep 1s
         }
-        local loggerBin; loggerBin=$(basename "$SOFLOGGER")
-        sudo pkill -9 "$loggerBin" 2>/dev/null || {
+
+        local loggerBin wcLog; loggerBin=$(basename "$SOFLOGGER")
+        # logfile is set in a different file: lib.sh
+        # shellcheck disable=SC2154
+        wcLog=$(wc -l "$logfile")
+        dlogi "nlines=$wcLog"
```
Comment (xiulipan) [inline case-lib/hijack.sh:38]: maybe a more readable log here? This looks like a cmd.

**Review by xiulipan — CHANGES_REQUESTED** [review]
Do we need any check if logger is not running before exiting?
This looks wired if test pass but logger failed and whole test case will exit with failure.

File: case-lib/hijack.sh [hunk 2]
```diff
@@ -30,12 +30,23 @@ function func_exit_handler()
             func_lib_start_log_collect 1
             sleep 1s
         }
-        local loggerBin; loggerBin=$(basename "$SOFLOGGER")
-        sudo pkill -9 "$loggerBin" 2>/dev/null || {
+
+        local loggerBin wcLog; loggerBin=$(basename "$SOFLOGGER")
+        # logfile is set in a different file: lib.sh
+        # shellcheck disable=SC2154
+        wcLog=$(wc -l "$logfile")
```
[the remark this answers is not in the document]
Comment (marc-hb) [reply case-lib/hijack.sh:37]: Good point, I'll move it right after killing the logger.

File: case-lib/hijack.sh (hunk 1 above)
Comment (marc-hb) [reply case-lib/hijack.sh:38]: In the next PR I will implement a sanity check on the number of lines. For now this is not great looking but it's readable and I'd rather not write and test code that will change in the next PR.

**Review by marc-hb — COMMENTED** [review]
Thanks @xiulipan !

> Do we need any check if logger is not running before exiting? This looks wired if test pass but logger failed and whole test case will exit with failure.

I don't understand this comment sorry. We already had for a few weeks  the "sof-logger was already dead" check that makes the test fail if the logger crashed. It works; I tested it before and after this PR.

**Comment by xiulipan** [comment]
@marc-hb Eve- though we had such failures in past, I want to start a discussion here: it that behavior right? Do we need to fail the case if we could not get the logger? As we have standalone test case for logger.

**Comment by marc-hb** [comment]
> it that behavior right? Do we need to fail the case if we could not get the logger? As we have standalone test case for logger.

The name of issue #297 is "Check for a valid trace for ALL tests" so I moved your question there.

**Comment by plbossart** [comment]
> @marc-hb Eve- though we had such failures in past, I want to start a discussion here: it that behavior right? Do we need to fail the case if we could not get the logger? As we have standalone test case for logger.

yes we want to fail if the logger is gone during a test. There could be something with a specific test that gets in the way of the DMA trace, and we want to trap this error.

**Review by xiulipan — APPROVED** [review]

**Pull request merged**